test/test_operations_problem.jl

```diff
@@ -171,52 +162,45 @@ end
         if network == StandardPTDFModel
             push!(LMPs, abs.(psi_ptdf_lmps(res, ptdf)))
         else
-            duals = res.dual_values[:nodal_balance_active__Bus]
+            duals = res.dual_values[:NodalBalanceActiveConstraint_Bus]
             duals = abs.(duals[:, propertynames(duals) .!== :DateTime])
             push!(LMPs, duals[!, sort(propertynames(duals))])
         end
     end
-
     @test isapprox(LMPs[1], LMPs[3], atol = 100.0)
 end
 
 @testset "Test ProblemResults interfaces" begin
     sys = PSB.build_system(PSITestSystems, "c_sys5_re")
-    template = get_template_dispatch_with_network(CopperPlatePowerModel)
-    model = DecisionModel(
-        template,
-        sys;
-        optimizer = OSQP_optimizer,
-        constraint_duals = [:CopperPlateBalance],
+    template = get_template_dispatch_with_network(
+        NetworkModel(CopperPlatePowerModel; duals = [CopperPlateBalanceConstraint]),
     )
+    model = DecisionModel(template, sys; optimizer = OSQP_optimizer)
     @test build!(model; output_dir = mktempdir(cleanup = true)) == PSI.BuildStatus.BUILT
     @test solve!(model) == RunStatus.SUCCESSFUL
 
     container = PSI.get_optimization_container(model)
-    constraints = PSI.get_constraints(container)[PSI.ConstraintKey(
-        CopperPlateBalanceConstraint,
-        PSY.System,
-    )]
-    dual_results = read_duals(container, [:CopperPlateBalance])
+    constraint_key = PSI.ConstraintKey(CopperPlateBalanceConstraint, PSY.System)
+    constraints = PSI.get_constraints(container)[constraint_key]
+    dual_results = read_duals(container)[:CopperPlateBalanceConstraint_System]
```

Comment:
@sourabhdalvi @jd-lara Similar to above, I was planning to have something like
```
dual_results = read_duals(container)[(CopperPlateBalanceConstraint, PSY.System)]
```

Reply:
forgetting the results I think this is fine, albeit inconsistent with the other interface but the dual interface will be for programmatic querying of the duals and it isn't too bad.

Reply:
Shouldn't the interfaces in the user code be symmetric? Otherwise, how will they know what to pass to `read_duals`?

To be clear I mean
```
NetworkModel(network; PTDF = ptdf, duals = (CopperPlateBalanceConstraint, PSY.System))
read_duals(container, (CopperPlateBalanceConstraint, PSY.System))
```

Reply:
I get what you mean 100% and agree that there is inconsistency. I am not sure what is worse that's all because in the case of the device and service model we don't need to pass the device type for the duals.